src/api/resources/sync/views.py

```diff
@@ -311,30 +343,18 @@ def vw_pull(request):
         exception = NotAuthenticated if 401 in status_codes else PermissionError
         raise exception(f"{str(exception)}: {failed_src_types}")
 
-    for source_type, source_data in request_data.items():
-        if source_type == CHOICES_SOURCE_TYPE:
-            response_data[source_type] = _get_choices()
-            continue
-
-        src = _get_source(source_type)
-
-        try:
-            req_params = _get_required_parameters(
-                request, source_data, src["required_filters"]
-            )
-        except ValueError as ve:
-            raise ValidationError(str(ve))
-
-        response_data.update(
-            {source_type: get_serialized_records(src["view"], **req_params)}
-        )
+    response_data = {
+        source_type: _get_source_records(source_type, source_data, request)
+        for source_type, source_data in request_data.items()
+    }
 
     return Response(response_data)
 
 
 @api_view(http_method_names=["POST"])
 def vw_push(request):
     request_data = request.data or {}
+    force = str(request.query_params.get("force")).lower() in ("1", "t", "true", "y", "yes")
```

Comment:
Don't we have a "truthy" util function somewhere? Also wonder if a `strip()` is in order.